### ISSUE TITLE
[Bug] - Plot mc_coordinates are stored from the player position not the plot center

### ISSUE THREAD

**Issue by MineFact** [AlpsBTE/Plot-System-Terra#4]
# Description
Currently the only way how to get the real terra coordinates of a plot is by using the "mc_coordinates" column in the database.
Another way is to analyze the schematic itself but thats not elegant.

The mc_coordinates data is used in the Plot System plugin to create the Google Maps links.
Its also used to recreate the schematic outline after completion, altough that isn't really necessary since the original schematic already contains the needed region outline.
In the Plot System Terra plugin its used to paste back the plot.

# Problem
The problem is that those mc_coordinates are from the player position after he scanned the project.
This could be far away from the plot itself which would make the Goole Maps links inaccurate.
The player position itself is also not worth to save since it doesn't provide any important information.

# Solution
Instead it would be much simpler to just save the plot center position and change the origin of the scanned region to the plot center too. This should not break any of the current usages since the origin is just somewhere else.

# Code
https://github.com/AlpsBTE/Plot-System-Terra/blob/5e5840cdeccc4d45ae0b2fcfc75aff25266a235a/src/main/java/com/alpsbte/plotsystemterra/core/plotsystem/PlotCreator.java#L134-L140

**Comment by LordTuxn** [comment]
This will also require few compatibility changes to the Plot-System plugin for plot saving/pasting/teleporting (tpll)

**Comment by MineFact** [comment]
No tpll uses the schematic region center to convert the coordinates.

Plot saving and plot pasting would not have to be changed if the origin of the scanned region is the center of the plot too.
Then the origin of the mc_coordinates location would be the same as the region origins

**Comment by MineFact** [comment]
Ok I tested it, everything still worked fine. The plot center is still at 75 75 and it gets pasted back at the correct location once finished and uploaded.
You can try it out if you want.

**Comment by LordTuxn** [comment]
> No tpll uses the schematic region center to convert the coordinates.
> 
> Plot saving and plot pasting would not have to be changed if the origin of the scanned region is the center of the plot too. Then the origin of the mc_coordinates location would be the same as the region origins

Yes, true. I looked at it again and there shouldn't really be any problems. 👍

**Comment by LordTuxn** [comment]
> Ok I tested it with 3 different plots, everything still worked fine. The plot center is still at 75 75 in the Plot World and it gets pasted back at the correct location once finished and uploaded. You can try it out if you want.

I will test it tomorrow

**Comment by MineFact** [comment]
thx 👍
